earth2mip/networks/__init__.py

```diff
@@ -233,7 +233,9 @@ def _iterate(self, x, normalize=True, time=None):
 
             while True:
                 if self.source:
-                    x = self.source(x, self.time_step)
+                    x_with_units = x * self.scale + self.center
+                    dt = torch.tensor(self.time_step.total_seconds()) / 86400.0
+                    x += self.source(x_with_units, time) / self.scale * dt
```

Comment:
Should we use SI units? (1/s)

```suggestion
                    x_with_units = x * self.scale + self.center
                    dt = torch.tensor(self.time_step.total_seconds()) 
                    x += self.source(x_with_units, time) / self.scale * dt
```